### PR TITLE
[1.13.2] Add getter method for MouseHelper.middleDown

### PR DESCRIPTION
**Pull request by Geforce132**
The current MouseHelper class has getters for leftDown and rightDown, which are booleans that return true if the player is holding down their left or right mouse buttons respectively. However, for some reason, MouseHelper doesn't have a getter method for middleDown, which is a boolean that checks whether or not the player's middle mouse button is being held down. This could be a useful method for mods that want to utilize middle-mouse to perform some function (including one of mine). 

An AccessTransformer or Reflection could also allow us access to this private value, but that's fairly impractical to have to set up an AT or deal with the performance issues of Reflection (if you're referencing this boolean a lot) when there's already getters for the other two mouse buttons. This commit simply adds a return method for middleDown into MouseHelper.

### CODE REVIEW

**Comment by williewillus** [comment]
Is there any other private data in this class we want to expose too? So we can get em all at once

**Comment by Geforce132** [comment]
I don't believe so? The only other private variables that I think would ever be used by any type of mod in MouseHelper is x/yVelocity, but I don't know why that would be needed.

**Comment by Cadiboo** [comment]
> The only other private variables that I think would ever be used by any type of mod in MouseHelper is x/yVelocity, but I don't know why that would be needed.

I can think of mods would render something (maybe an entity or model) looking in direction and have it moveable with click & drag, which will probably use velocity

**Comment by bl4ckscor3** [comment]
Merging this would be very helpful for ease of accessibility.

File: patches/minecraft/net/minecraft/client/MouseHelper.java.patch [hunk 1]
```diff
@@ -40,3 +40,14 @@
              }, "mouseDragged event handler", iguieventlistener.getClass().getCanonicalName());
           }
  
+@@ -224,6 +232,10 @@
+       return this.field_198039_d;
+    }
+ 
++   public boolean isMiddleDown() {
```
Comment (mezz) [inline patches/minecraft/net/minecraft/client/MouseHelper.java.patch:56]: Please put new methods at the bottom of the file, after a comment like this:
` /* ======================================== FORGE START =====================================*/`

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by mezz** [comment]
Superseded by https://github.com/MinecraftForge/MinecraftForge/pull/5776

**Pull request closed without merging**